Mappers/HttpMapper.php

```diff
@@ -194,6 +241,22 @@ protected function getHttpClient()
         return $this->httpClient;
     }
 
+    /**
+     * @param array $data
+     * @param string $method
+     *
+     * @return ArraySerializable
+     *
+     * @throws InternalException
+     * @throws \Infrastructure\Models\Http\Response\ResponseContentTypeException
+     */
+    private function updateByHttpMethod(array $data, string $method): ArraySerializable
```

Comment:
Maybe instead of creation new  method you will move 
 $this->requestFactory->create()
inside $this->sendRequestForEntity() ? 
Or you can avoid query chain for $this->requestFactory->create() by creating moethod only for createRequest() ?